pkg/core/interop/runtime/util.go

```diff
@@ -88,9 +89,26 @@ func GetNetwork(ic *interop.Context) error {
 
 // GetRandom returns pseudo-random number which depends on block nonce and transaction hash.
 func GetRandom(ic *interop.Context) error {
-	res := murmur128(ic.NonceData[:], ic.Network)
+	var (
+		price int64
+		seed  = ic.Network
+	)
+	isHF := ic.IsHardforkEnabled(config.HFAspidochelone)
+	if isHF {
+		price = 1 << 13
+		seed += ic.GetRandomCounter
+		ic.GetRandomCounter++
+	} else {
+		price = 1 << 4
+	}
+	res := murmur128(ic.NonceData[:], seed)
+	if !isHF {
+		copy(ic.NonceData[:], res)
+	}
+	if !ic.VM.AddGas(ic.BaseExecFee() * price) {
```

Comment:
@roman-khimov, it may be a nitpicking, but I'd better add GAS firstly and only after that compute random and update context nonce. Otherwice, the node may work for free if there's not enough GAS.

Current implementation matches the C# one.